libs/backend/src/list_directory.ts

```diff
@@ -110,6 +110,35 @@ export async function listDirectory(
   }
 }
 
+/**
+ * Get entries for a directory recursively, includes stat information for each entry.
+ * Requires that the path be absolute. Includes directories in result.
+ */
+export async function listDirectoryRecursive(
+  path: string
+): Promise<Result<FileSystemEntry[], ListDirectoryError>> {
+  const listRootResult = await listDirectory(path);
+  if (listRootResult.isErr()) {
+    return err(listRootResult.err());
+  }
```

Comment:
As in my generator-based version, you can just do this:

```suggestion
  if (listRootResult.isErr()) {
    return listRootResult;
  }
```

Wishing for this:

```rust
let file_entries = list_directory(path)?;
```

✨